src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -595,7 +596,16 @@ fn env_args(
         return Ok(Vec::new());
     }
 
-    // First try RUSTFLAGS from the environment
+    // First try CARGO_ENCODED_RUSTFLAGS from the environment.
+    // Prefer this over RUSTFLAGS since it's less prone to encoding errors.
+    if let Ok(a) = env::var(format!("CARGO_ENCODED_{}", name)) {
+        if a.is_empty() {
+            return Ok(Vec::new());
+        }
```

Comment:
This doesn't seem necessary?

```suggestion
```

Reply:
Ah, it is, because `.split()` on an empty string returns an iterator with one item (`""`), which we'd pass as an empty argument to rustc, which can make it confused. See also https://github.com/rust-lang/cargo/pull/9601#discussion_r661037078.